python/cudf/cudf/core/column/column.py

```diff
@@ -1346,9 +1346,33 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             arbitrary = arbitrary.astype(dtype)
 
         if arb_dtype.kind == "M":
-            data = as_column(pa.array(arbitrary), nan_as_null=nan_as_null)
-            if np.issubdtype(data.dtype, np.datetime64):
+            import datetime as dt
+
+            time_unit, _ = np.datetime_data(arbitrary.dtype)
+            cast_dtype = time_unit in ("D", "W", "M", "Y") or (
+                len(arbitrary) > 0
+                and (
+                    isinstance(arbitrary[0], str)
+                    or isinstance(arbitrary[0], dt.datetime)
```

Comment:
It looks like these are for handling an `object` array of strings or Python datetime objects. I think we can just remove these.

Reply:
Removed this check for array of strings/datetime objects.